scripts/lib/CIME/case/case_clone.py

```diff
@@ -39,7 +39,10 @@ def create_clone(self, newcaseroot, keepexe=False, mach_dir=None, project=None,
         logger.warning(" It is NOT recommended to clone cases from different versions of CIME.")
 
     # *** create case object as deepcopy of clone object ***
-    srcroot = os.path.join(newcase_cimeroot,"..")
+    if os.path.isdir(os.path.join(newcase_cimeroot,'share')) and get_model() == "cesm":
+        srcroot = newcase_cimeroot         
```

Comment:
I don't feel a need to hold up this PR for this comment, but I'll make it here and we can open a separate issue if we don't want to hold up the PR:

I'm concerned about this logic. It seems very possible that the share directory would exist within cime, because someone ran manage_externals from within cime, either by accident or on purpose, yet the user's intent is to use a srcroot one level up from cime. It seems like most/all "normal" users will be using cime in the situation where srcroot is one level up from cimeroot, and so that case should always be the default. The situation where you're running create_clone from a standalone cime checkout can be handled the same way as what I think is done for create_newcase: requiring a `--srcroot` argument. I realize that something still would need to be done for the sake of system tests.

Here's an email I sent about 6 weeks ago; I'm wondering if this would work or if there are problems with this suggestion:

> Maybe you've already seen this, but in case not: I poked around for a few minutes and found this:
> 
> https://github.com/esmci/cime/blob/master/scripts/create_newcase#L101-L108
> 
> However, it looks like there isn't a corresponding --srcroot argument to create_test. I'm thinking that, if you added a --srcroot argument to create_test then that would get you part of the way.
> 
> But this would also be a problem, since some tests use create_clone:
> 
> https://github.com/esmci/cime/blob/master/scripts/lib/CIME/case/case_clone.py#L42
> 
> I'm thinking there are actually a few problems there:
> 
> (1) For user use, create_clone should have the same srcroot logic as create_newcase, with the ability to override it just like in create_newcase. (This isn't important for getting your testing to work, but would be a good thing in general.)
> 
> (2) Maybe for the purposes of system tests, the create_clone function needs an option like keep_original_srcroot; when called from the system tests, that should be set to true, in which case the clone's SRCROOT is set equal to the original case's SRCROOT.
> 
> With those things in place, I'm thinking you'd be able to get this to work if you specify '--srcroot ..' (assuming you're running this from within cime/scripts) to the create_test command.
>

Reply:
I am deferring this to its own issue: #3923